evap/staff/importers.py

```diff
@@ -97,28 +99,87 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
         assert not self.errors
         # This is safe because the user's email address is checked before in the importer (see #953)
         responsible_dbobj = UserProfile.objects.get(email=self.responsible_email)
-        course = Course(
-            name_de=self.name_de,
+        if self.existing_course:
+            self.existing_course.degrees.add(*self.degrees)
+        else:
+            course = Course(
+                name_de=self.name_de,
+                name_en=self.name_en,
+                type=self.course_type,
+                semester=semester,
+            )
+            course.save()
+            course.responsibles.set([responsible_dbobj])
+            course.degrees.set(self.degrees)
+            evaluation = Evaluation(
+                vote_start_datetime=vote_start_datetime,
+                vote_end_date=vote_end_date,
+                course=course,
+                wait_for_grade_upload_before_publishing=self.is_graded,
+            )
+            evaluation.save()
+            evaluation.contributions.create(
+                evaluation=evaluation,
+                contributor=responsible_dbobj,
+                role=Contribution.Role.EDITOR,
+                textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+            )
+
+    def check_existing_course(self, semester, importer):
```

Comment:
I am not sure how we should handle the generation of error messages. Putting those into `check_evaluation_data_correctness` would make that method extremely long. Generating those error messages outside of it needs the importer object and hides the places where the error messages get generated.
Another solution could be writing the error messages into `EvaluationData.errors` and convert them to importer errors later on like here: (`importers.py` Line 501-509)
```python
if "is_graded" in evaluation_data.errors:
      self.errors[ImporterError.IS_GRADED].append(
          _('"is_graded" of course {} is {}, but must be {} or {}').format(
              evaluation_data.name_en,
              evaluation_data.errors["is_graded"],
              settings.IMPORTER_GRADED_YES,
              settings.IMPORTER_GRADED_NO,
          )
      )
```

Reply:
Hm, I feel like we shouldn't make this super complicated because error generation will move around in / after #1634 anyways. I think all of your solutions are fine and should work, I think that its okay to pass the importer into these methods here

... but, @He3lixxx probably has the most up-to-date opinions on the matter for reasons not disclosed here

Reply:
In case we merge #1634, structuring of code in this PR wouldn't matter, so then you wouldn't have to worry too much. If you want to think about useful structuring of importer code, maybe take a look at that PR, suggestions are welcome there.

Reply:
Ok, so I will leave it as is :)